compiler/plugins/target/ROCM/Dialect/ROCM/Transforms/ApplyBuiltinPDLPatterns.cpp

```diff
@@ -251,4 +263,86 @@ class ApplyBuiltinPDLPatternsPass
   FrozenRewritePatternSet patterns;
 };
 
+class ApplyBuiltinPDLPatternsDriverPass final
+    : public mlir::iree_compiler::IREE::ROCM::impl::
+          ApplyBuiltinPDLPatternsDriverPassBase<
+              ApplyBuiltinPDLPatternsDriverPass> {
+public:
+  using Base::Base;
+
+  void runOnOperation() override {
+    ModuleOp moduleOp = getOperation();
+    IREE::GPU::TargetAttr targetAttr = getGPUTargetAttr(moduleOp);
+    if (!targetAttr) {
+      moduleOp->emitOpError() << "no executable target attribute found.";
+      return signalPassFailure();
+    }
+
+    IREE::ROCM::ApplyBuiltinPDLPatternsPassOptions options;
+    options.enableTensorUKernels = true;
+    options.targets.push_back(targetAttr.getArch().str());
+    OpPassManager passManager(moduleOp.getOperationName());
+    FunctionLikeNest(passManager).addPass([&]() {
+      return createApplyBuiltinPDLPatternsPass(options);
+    });
+    if (failed(runPipeline(passManager, moduleOp))) {
+      signalPassFailure();
+    }
+
+    // Parse the required builtin ukernels and embed them into the module.
+    MLIRContext *ctx = moduleOp.getContext();
+    auto rocmDialect = ctx->getOrLoadDialect<IREE::ROCM::ROCMDialect>();
+    SmallVector<IREE::Util::FuncOp> ukernelFunctions;
+    auto res = moduleOp.walk([&](Operation *op) {
+      auto builtinName =
+          dyn_cast_or_null<StringAttr>(op->getAttr(kBuiltinName));
+      auto ukernelDesc = dyn_cast_or_null<IREE::Codegen::UKernelDescriptorAttr>(
+          op->getAttr(kUKernelDescriptorName));
```

Comment:
`IREECodegenAttrs.h` has a `get/setUKernelDescriptor` helper for this so we don't need to expose the internal attribute name.